akka-http/src/main/scala/akka/http/scaladsl/server/directives/BasicDirectives.scala

```diff
@@ -312,7 +357,7 @@ object BasicDirectives extends BasicDirectives {
   private val _extractLog: Directive1[LoggingAdapter] = extract(_.log)
   private val _extractSettings: Directive1[RoutingSettings] = extract(_.settings)
   private val _extractParserSettings: Directive1[ParserSettings] = extract(_.parserSettings)
-  private val _extractRequestContext: Directive1[RequestContext] = extract(conforms)
+  private val _extractRequestContext: Directive1[RequestContext] = extract(identity)
```

Comment:
conforms may be deprecated however it does not allocate new functions :-)
Please use this instead: `akka.stream.impl.ConstantFun.scalaIdentityFunction` (it's a performance optimisation)